test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/survival/KieServerWithWorkbenchSurvivalIntegrationTest.java

```diff
@@ -0,0 +1,164 @@
+
+public class KieServerWithWorkbenchSurvivalIntegrationTest {
+
+    private static final String ORGANIZATION_UNIT_NAME = "myOrgUnit";
+    private static final String REPOSITORY_NAME = "myRepo";
+
+    private static final String PROJECT_GROUP_ID = "org.kie.server.testing";
+    private static final String PROJECT_NAME = "definition-project";
+    private static final String PROJECT_VERSION = "1.0.0.Final";
+
+    private static final String CONTAINER_ID = "cont-id";
+    private static final String CONTAINER_ALIAS = "cont-alias";
+
+    private static final String SIGNALTASK_PROCESS_ID = "definition-project.signaltask";
+
+    private static final DeploymentScenarioBuilderFactory deploymentScenarioFactory = DeploymentScenarioBuilderFactory.getInstance();
+    private WorkbenchWithKieServerScenario workbenchWithKieServerScenario;
+    private WorkbenchClientProvider workbenchClientProvider;
+    private KieServerClientProvider kieServerClientProvider;
+    private KieServerControllerClientProvider kieServerControllerClientProvider;
+    private KieServerMgmtControllerClient kieServerMgmtControllerClient;
+    private GitProvider gitProvider;
+
+    private static final Logger logger = LoggerFactory.getLogger(KieServerWithWorkbenchSurvivalIntegrationTest.class);
+
+    @Before
+    public void setUp() {
+        workbenchWithKieServerScenario = deploymentScenarioFactory.getWorkbenchWithKieServerScenarioBuilder().build();
+        workbenchWithKieServerScenario.deploy();
+
+        gitProvider = GitProviderFactory.getGitProvider();
+        gitProvider.createGitRepository(workbenchWithKieServerScenario.getNamespace(), ClassLoader.class.getResource("/kjars-sources").getFile());
+
+        workbenchClientProvider = new WorkbenchClientProvider(workbenchWithKieServerScenario.getWorkbenchDeployment());
+        workbenchClientProvider.createOrganizationalUnit(ORGANIZATION_UNIT_NAME, workbenchWithKieServerScenario.getWorkbenchDeployment().getUsername());
+        workbenchClientProvider.cloneRepository(ORGANIZATION_UNIT_NAME, REPOSITORY_NAME, gitProvider.getRepositoryUrl(workbenchWithKieServerScenario.getNamespace()));
+        workbenchClientProvider.deployProject(REPOSITORY_NAME, PROJECT_NAME);
+
+        kieServerClientProvider = new KieServerClientProvider(workbenchWithKieServerScenario.getKieServerDeployment());
+
+        kieServerControllerClientProvider = new KieServerControllerClientProvider(workbenchWithKieServerScenario.getWorkbenchDeployment());
+        kieServerMgmtControllerClient = kieServerControllerClientProvider.getKieServerMgmtControllerClient();
+        kieServerControllerClientProvider.waitForServerTemplateCreation();
+    }
+
+    @After
+    public void tearDown() {
+        InstanceLogUtil.writeDeploymentLogs(workbenchWithKieServerScenario.getWorkbenchDeployment());
+        InstanceLogUtil.writeDeploymentLogs(workbenchWithKieServerScenario.getKieServerDeployment());
+        workbenchWithKieServerScenario.undeploy();
```

Comment:
Please add here if (.. != null) check to prevent NPE when for example gitProvider is null. For example see https://github.com/kiegroup/kie-cloud-tests/pull/22